tika-translate/src/main/java/org/apache/tika/language/translate/Lingo24Translator.java

```diff
@@ -43,72 +46,78 @@
-        this.isAvailable = true;
-        Properties config = new Properties();
-        try {
-            config.load(Lingo24Translator.class
-                    .getResourceAsStream(
-                            "translator.lingo24.properties"));
-            this.userKey = config.getProperty("translator.user-key");
-            if (this.userKey.equals(DEFAULT_KEY))
-                this.isAvailable = false;
-        } catch (Exception e) {
-            e.printStackTrace();
-            isAvailable = false;
-        }
-    }
-
-    @Override
-    public String translate(String text, String sourceLanguage,
-                            String targetLanguage) throws TikaException, IOException {
-        if (!this.isAvailable)
-            return text;
-        Response response = client.accept(MediaType.APPLICATION_JSON)
-                .query("user_key", userKey).query("source", sourceLanguage)
-                .query("target", targetLanguage).query("q", text).get();
-        BufferedReader reader = new BufferedReader(new InputStreamReader(
-                (InputStream) response.getEntity(), UTF_8));
-        String line = null;
-        StringBuffer responseText = new StringBuffer();
-        while ((line = reader.readLine()) != null) {
-            responseText.append(line);
-        }
-
-        ObjectMapper mapper = new ObjectMapper();
-        JsonNode jsonResp = mapper.readTree(responseText.toString());
-        if (jsonResp.findValuesAsText("errors").isEmpty()) {
-            return jsonResp.findValuesAsText("translation").get(0);
-        } else {
-            throw new TikaException(jsonResp.findValue("errors").get(0).asText());
-        }
-    }
-
-    @Override
-    public String translate(String text, String targetLanguage)
-            throws TikaException, IOException {
-        if (!this.isAvailable)
-            return text;
-        LanguageIdentifier language = new LanguageIdentifier(
-                new LanguageProfile(text));
-        String sourceLanguage = language.getLanguage();
-        return translate(text, sourceLanguage, targetLanguage);
-    }
-
-    @Override
-    public boolean isAvailable() {
-        return this.isAvailable;
-    }
+	private static final String LINGO24_TRANSLATE_URL_BASE = "https://api.lingo24.com/mt/v1/translate";
+
+	private static final String DEFAULT_KEY = "dummy-key";
+
+	private String userKey;
+
+	public Lingo24Translator() {
+		Properties config = new Properties();
+		try {
+			config.load(Lingo24Translator.class
+					.getResourceAsStream(
+							"translator.lingo24.properties"));
+			this.userKey = config.getProperty("translator.user-key");
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+
+	@Override
+	public String translate(String text, String sourceLanguage,
+			String targetLanguage) throws TikaException, IOException {
+		if (!this.isAvailable()) {
+			return text;
+		}
+		WebClient client = WebClient.create(LINGO24_TRANSLATE_URL_BASE);
```

Comment:
WebClient is not thread-sfe and was declared as instance variable before, after the first request it would actually append query parameters to the already defined parameters for each subsequent request.
For instance, on the third request to the server it would regerate a request with URL like this: https://.....?source=es&source=es&source=es......